oai_agents/agents/rl.py

```diff
@@ -300,26 +300,26 @@ def train_agents(self, total_train_timesteps, exp_name=None):
             self.ck_list.append(({k: 0 for k in self.args.layout_names}, path, tag))
```

Comment:
You should ensure that the new ck_list starts from the previous ck not 0

Reply:
Is this stored in the agent somewhere? not sure where to get the original value from since it seems like "agent" is just the sb3 agent

Reply:
Or do i need to save this as well, in the case of last i cant look at the name

Reply:
this is stored in the directory under args.exp_dir. look at this thread for reference: https://cu-hiro.slack.com/archives/C07CA2AKQDB/p1728063937748989?thread_ts=1727985882.154229&channel=C07CA2AKQDB&message_ts=1728063937.748989

Reply:
so essentially:
when args.resume = True:
  skip this if statement (lines 297-300): [link](https://github.com/HIRO-group/multiHRI/blob/2552629130f9cb672678747a6492a076221cdd7c/oai_agents/agents/rl.py#L297)
  Makes sure that ck_list is indexed properly here: [link](https://github.com/HIRO-group/multiHRI/blob/2552629130f9cb672678747a6492a076221cdd7c/oai_agents/agents/rl.py#L335)


 In order to make sure the ck_list is indexed properly, you should construct the ck_list: `self.ck_list.append((rew_per_layout, path, tag))` by reading the directory under `args.exp_dir/agent.name`. One way to implement this would be to change this [line](https://github.com/HIRO-group/multiHRI/blob/2552629130f9cb672678747a6492a076221cdd7c/oai_agents/agents/rl.py#L297) to be something like this:

```
if self.checkpoint_rate:
   if args.resume==False:
            self.ck_list = []
            path, tag = self.save_agents(tag=f'ck_{len(self.ck_list)}')
            self.ck_list.append(({k: 0 for k in self.args.layout_names}, path, tag))
   elif args.resume==True:
            self.ck_list = self.construct_ck_list()
```
and construct_ck_list() should read args.exp_dir/agent.name to create ck_list from file.

Reply:
Should we be passing in which checkpoint to use as the start for restart then? Might be better than using last unless last can only be saved at the same time as the other checkpoint dir

Reply:
We don't have to pass it, it should automatically be read from the directory under `args.exp_dir/agent.name` and select the last `ck_[i]_rew` to restart from. We cannot use `last` because `last` is only created when the training finishes and in case of a crash it won't be created